browser/android/youtube_script_injector/brave_youtube_script_injector_native_helper_browsertest.cc

```diff
@@ -0,0 +1,306 @@
+
+  EXPECT_TRUE(IsVideoPlaying());
+  // Wait for the resize to complete triggered by fullscreen change.
+  content::WaitForResizeComplete(web_contents());
+
+  // Check the video is in fullscreen mode.
+  EXPECT_TRUE(
+      WaitForJsResult(web_contents(), "document.fullscreenElement !== null"));
+
+  // Check again the video is still playing.
+  EXPECT_TRUE(IsVideoPlaying());
+}
+
+IN_PROC_BROWSER_TEST_F(BraveYouTubeScriptInjectorNativeHelperBrowserTest,
+                       VideoIsPlayedFromFullscreenState) {
+  const GURL url = https_server_.GetURL("youtube.com", "/yt_fullscreen.html");
+  content::NavigateToURLBlockUntilNavigationsComplete(web_contents(), url, 1,
+                                                      true);
+
+  // Video is not playing.
+  ASSERT_FALSE(IsVideoPlaying());
+  // Simulate fullscreen button click to enter fullscreen mode.
+  ASSERT_TRUE(content::ExecJs(
+      web_contents(),
+      "document.querySelector('button.fullscreen-icon').click()"));
+  // Wait for the resize to complete triggered by fullscreen change.
+  content::WaitForResizeComplete(web_contents());
+  // Assert the video is in fullscreen mode.
+  ASSERT_TRUE(
+      WaitForJsResult(web_contents(), "document.fullscreenElement !== null"));
+
+  youtube_script_injector::SetFullscreen(web_contents());
+
+  EXPECT_TRUE(WaitForJsResult(
+      web_contents(),
+      "document.querySelector('video.html5-main-video').paused === false"));
+}
+
+IN_PROC_BROWSER_TEST_F(BraveYouTubeScriptInjectorNativeHelperBrowserTest,
+                       NoOpIfPlayerIsNotFound) {
+  // Load a page without a video.
+  const GURL url = https_server_.GetURL("youtube.com", "/ytcftg_mock.html");
+  content::NavigateToURLBlockUntilNavigationsComplete(web_contents(), url, 1,
+                                                      true);
+
+  // Assert that movie player is not present.
+  ASSERT_TRUE(
+      content::EvalJs(web_contents(),
+                      "document.getElementById('movie_player') === null")
+          .ExtractBool());
+
+  std::string dom_before =
+      content::EvalJs(web_contents(), "document.body.innerHTML")
+          .ExtractString();
+  // Attempt to set fullscreen, which should not change anything.
+  youtube_script_injector::SetFullscreen(web_contents());
+
+  std::string dom_after =
+      content::EvalJs(web_contents(), "document.body.innerHTML")
+          .ExtractString();
+  // Assert that the DOM remains unchanged.
+  EXPECT_EQ(dom_before, dom_after);
+}
+
+IN_PROC_BROWSER_TEST_F(BraveYouTubeScriptInjectorNativeHelperBrowserTest,
+                       SetFullscreenOnElementsLoadingDelayed) {
+  const GURL url =
+      https_server_.GetURL("youtube.com", "/yt_fullscreen_delayed.html");
+  content::NavigateToURLBlockUntilNavigationsComplete(web_contents(), url, 1,
+                                                      true);
+
+  // Simluate delayed script load by presenting a player ('movie_player') that
```

Comment:
what do you mean "delayed script loading"? What is the source of the delay?

Reply:
looks like it shouldn't show up until the page is loaded https://github.com/brave/brave-core/pull/29431/files

Reply:
Improved test description 👉 [`154b4e6` (#29431)](https://github.com/brave/brave-core/pull/29431/commits/154b4e679af0f3e1be347d7d3eb0bb8abe4df010).